Revise wishlist content and wording (#184)
framedsc/Sitesource: markdown/the_photomode_wishlist.md

<p><i>Alan Wake II</i>, shot by seymourglass</p>
</div>

> This article is aimed at developers who are curious about what the "perfect" photomode looks like to players. Features cost resources, and the cost of implementing the ones listed in here can vary a lot, so this list is organized by priority.

Author: ```suggestion
This article is aimed at developers who are curious about what the "perfect" photomode looks like to players. Features take resources, and the effort of implementing the ones listed in here can vary, so this list is organized by priority.
```

I tried to avoid referencing the "cost" in terms of money and put it in terms of effort — seems like it might be an easier sell that way.

- Camera tilt
  : It is often overlooked, but part of the complete movement of a free camera is being able to tilt it (also called roll depending on the bibliography). Such movement can help a lot in making compositions feel more dynamic or allow players to shoot in a vertical aspect ratio (although being able to [hotsample to the desired aspect ratio](basics.htm#composing-for-custom-aspect-ratios) is much preferred).

- No camera clipping
  : Having the camera be free to move through geometry can be very useful, as it allows the player to position the camera in those places where it makes the most sense for the shot instead of only in the areas where the photomode allows them to.

- Photomode settings toggle
  : Pretty obvious considering we want to shoot the game itself without any menu getting in the way. Even if your game has a feature that automatically outputs the screenshot into a file, giving the player the option to take a shot of the game with their preferred software would also be a nice perk. Additionally, avoid rendering any kind of watermark in the photomode; they get in the way and end up being very distracting to the overall photo.

### Field-of-view

> Being able to control the "angle of the lens" of the camera is very important, but sometimes these have annoying quirks. Try to allow the minimum and maximum field of view that the engine supports, and provide enough granularity in the user interface so the player can choose the right FoV for their shot. Consider the real-world analogue to this control too: smaller values should give a wider view, larger values should be more narrow.

Reviewer: Do you mean that you want the values on the pm to mimic a real lens instead of the fov values? I am not sure if I am into that. I understand you want to draw a 1:1 with photography, but I believe it just adds an unnecessary layer of abstraction.

Author: You mean the last sentence there? It's not so much about having the FoV values match to lens millimeter numbers or anything, but for players with experience with cameras it wouldd be less confusing for to move an FoV slider down to give a wider view and up to get a narrow view.

Reviewer: I personally believe that there is a higher chance of players being more familiar with how FoV values work because of settings on FPS games than players that have ever used an irl camera in their lives. That plus the fact that I am more familiar with game engines than the average player means I am clearly biased, but that's how I see it.

Author: Ah I didn't realise that was a common setting in FPS games — we can just drop that mention completely then.

The following aspects regarding camera movement and orientation are crucial:

- Maximum range as big as possible

>   : Maybe your game or engine is not prepared to deal with the camera being far away from the player, but try to push this value as much as you can. Small range limits are one of the things that get modded first on PC, so players will certainly appreciate it. Some studios may be worried about players moving the camera around to find secrets or look at what comes next in the level, but this is a decision that comes from the player and in a photomode context the advantages outweigh the downsides.

Author: Might just be my ignorance of the gaming industry, but would the studio execs really care about players using photomodes to "peek ahead"?

I guess their angle might be more about seeing camera clipping or unrendered areas or something and thinking that reflects poorly on the product/brand though... 🤷

Reviewer: Sorry, I should have clarified why I changed this back. Yeah, I believe you are right in that game/level designers might care more about this than execs, and I am sure execs def care about clipping and taking the camera where we shouldn't. Thought that just writing studios might include all of the potential people that *might* have a problem with this, but I don't have a strong opinion about it, so feel free to ignore it.

Reviewer 2: I'd go for studios, as 'team' isn't really a term used when a game is released.

Author: I'm sure I've heard "team" used before — "The team behind such-and-such" or "The team at whatever studio" — but I don't mean to belabor the point too much. "Studio" is fine.

Or maybe "stakeholders" 😉

Letting the player spawn their own lights to re-illuminate the characters and scene can be a game changer. The controls and UI can be tricky to get right, but we suggest looking at what Insomniac has done with the Spider-Man games and Ratchet & Clank: Rift Apart. And, of course, what Frans has been doing with the [UUU](https://opm.fransbouma.com/uuuv5.htm#lights).

Besides letting the player spawn lights, it's almost equally important to let them turn off the in-game lights. That way they can start lighting the scene from scratch without having to deal with the other lights that don't benefit their composition.

> Of all of the features listed in this article, this one might be one of the most expensive ones to make. But, as the example below shows, being able to turn off the in-game lights and relit a scene by yourself might be one of the most transformative tools screenshoters have.

Author: ```suggestion
Of all of the features listed in this article, this one likely requires more effort to implement. But, as the example below shows, being able to turn off the in-game lights and re-light a scene completely is one of the most transformative tools screenshotters have.
```

A couple of typos, and reworded a bit to be a bit more "direct".

Reviewer: Lol sorry, I have been so exposed to "lit" tools that the correct wording has totally been forgotten 😆

### Post-processing

Besides the effects we mentioned above that we would like to get rid of (or at least have control over), there are also other post-processing effects that we might not want to delete completely, such as bloom or color correction. Having control over engine-specific post-processing settings is also useful if the resources allow them.

### Aspect ratio options

Players may not be familiar with the concept of aspect ratios, nor how a 16:9 screen is not a common ratio in photography, and are therefore restricted by what games have to offer. Other aspect ratios such as 3:4 or 2:1 can be incredibly beneficial to what they are trying to achieve.

> Ideally, these could be achieved by rendering the final image in the desired ratio, rather than with black bar overlays like [how we do it on PC](basics.htm#composing-for-custom-aspect-ratios). But understanding the limitations consoles have, black bar overlays might still be useful.

Author: ```suggestion
Ideally, these could be achieved by rendering the final image in the desired ratio — like [how it's done on PC](basics.htm#composing-for-custom-aspect-ratios) — rather than with black bar overlays. But understanding the limitations consoles have, black bar overlays might still be useful.
```

Reworded a bit to reduce confusion about what the PC method is.

Having the option to adjust the shutter speed, even in non-racing games, can be useful for many reasons, one of which is allowing the user to take long exposure photos (like trails of cars passing by). This may require an option to unpause the game while still being in photomode for this to work. Also consider introducing some kind of slow motion to have extra frames to work with. Otherwise, movement can look choppy.

> Decreasing shutter speed can have other benefits specific to games; see [our article on long exposure in ReShade](ReshadeGuides/RealLongExposure.htm) as it has more details.

Author: ```suggestion
Decreasing shutter speed can have other benefits specific to games; see [our article on long exposure in ReShade](ReshadeGuides/RealLongExposure.htm) for more details.
```

Matches the wording for a similar callout in the previous section.
